### PR TITLE
rfc: add share rfc

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

docs: add share rfc

Fixes https://github.com/datafuselabs/databend/issues/7058

### CODE REVIEW

File: docs/doc/60-contributing/03-rfcs/20220809-share.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+```
+CREATE DATABASE db FROM SHARE t
+```
+
+
+
+
+
+
+
```
Comment (Xuanwo) [inline docs/doc/60-contributing/03-rfcs/20220809-share.md:210]: Remove empty lines

File: docs/doc/60-contributing/03-rfcs/20220809-share.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+After granting the object privileges operations, can add tenants to the share.
+
+
+
+### Step 3: Add Tenants to the Share
+
+The following example adds two tenants to the t share:
+
+```SQL
```
Comment (Xuanwo) [inline docs/doc/60-contributing/03-rfcs/20220809-share.md:189]: use sql.

File: docs/doc/60-contributing/03-rfcs/20220809-share.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+Revoke access to databases and other supported database objects (tables) in the share. Revoking permissions to these objects effectively removes them from the share, making them inaccessible to all tenants that have created databases in the share.
+
+
+
+Syntax:
+
+```SQL
+REVOKE object_privilege ON grant_share_object_name FROM SHARE <share_name>
+
```
Comment (Xuanwo) [inline docs/doc/60-contributing/03-rfcs/20220809-share.md:100]: Seems wrong section.

File: docs/doc/60-contributing/03-rfcs/20220809-share.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+Describes creating and managing shares for tenants, and sharing shares with other tenants.
+
+## DDL for Shares
+
+To support creating and managing shares, Databend provides the following set of special DDL commands:
+
+* Create Share
+* Drop Share
+* Alter Share
+* Grant share_object privilege TO Share
```
Comment (Xuanwo) [inline docs/doc/60-contributing/03-rfcs/20220809-share.md:21]: Please define share object.

File: docs/doc/60-contributing/03-rfcs/20220809-share.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+### Drop Share
+
+Remove the specified share from the system and immediately revoke access to all tenants (i.e., the tenants that created the database from that share).
+
+
+
+Syntax:
+
+```SQL
+DROP SHARE [IF EXISTS] <share_name>`
```
Comment (Xuanwo) [inline docs/doc/60-contributing/03-rfcs/20220809-share.md:51]: There is a typo.

File: docs/doc/60-contributing/03-rfcs/20220809-share.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
+---
+title: Share objects
+description: Share objects between tenants
+---
+
+- Tracking Issue: [[Feature: tracing issue of share SQL support · Issue #6973 · datafuselabs/databend](https://github.com/datafuselabs/databend/issues/6973)
```
Comment (Xuanwo) [inline docs/doc/60-contributing/03-rfcs/20220809-share.md:6]: Markdown seems wrong

File: docs/doc/60-contributing/03-rfcs/20220809-share.md [hunk 7]
```diff
@@ -0,0 +1,210 @@
+---
+title: Share objects
+description: Share objects between tenants
```
Comment (Xuanwo) [inline docs/doc/60-contributing/03-rfcs/20220809-share.md:3]: Please define tenant first.

Comment (BohuTANG) [reply docs/doc/60-contributing/03-rfcs/20220809-share.md:3]: The document preview address is: https://databend-git-fork-lichuang-addsharerfc-databend.vercel.app/doc/contributing/rfcs/share

**Comment by Xuanwo** [comment]
FYI: This PR's type is `rfc`, which requires three approval to be merged.

File: docs/doc/60-contributing/03-rfcs/20220809-share.md [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+```sql
+SHOW GRANTS
+SHOW GRANTS ON grant_share_object_name
+SHOW GRANTS TO {TENANT <tenant_name> | SHARE <share_name> }
+SHOW GRANTS OF SHARE <share_name>
+
+grant_share_object_name := { DATABASE <db_name> | TABLE <db_name>.<table_name> }
+```
+
+- `SHOW GRANTS`: Syntactically equivalent to SHOW GRANTS TO TENANT current_tenant. Lists all the roles granted to the current tenant.
```
Comment (drmingdrmer) [inline docs/doc/60-contributing/03-rfcs/20220809-share.md:107]: What is the meaning of term `role`?

File: docs/doc/60-contributing/03-rfcs/20220809-share.md [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+- `SHOW GRANTS ON grant_share_object_name`: Lists all privileges that have been granted on the object.
+- `SHOW GRANTS TO ...`:
+  - `TENANT <tenant_name>`: Lists all the privileges granted to the tenant.
+  - `SHARE <share_name>`: Lists all the privileges granted to the share.
```
Comment (drmingdrmer) [inline docs/doc/60-contributing/03-rfcs/20220809-share.md:111]: Is the term `privilege` the same as `object_privilege` at line-87?

File: docs/doc/60-contributing/03-rfcs/20220809-share.md [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+### Step 2: Grant Privileges for a Database and Objects to the Share
+
+To include objects in a share, grant permissions to each object. When granting permissions, first grant usage rights to any container objects before granting usage rights to the objects in the container. For example, grant access to the database before granting access to any table included in the database.
+
+The following example illustrates granting privileges on the following objects to the t share created in the previous step:
+
+```sql
+grant usage on database db1 to share t;
+
+grant select on table db1.table1 to share t;
```
Comment (drmingdrmer) [inline docs/doc/60-contributing/03-rfcs/20220809-share.md:150]: Is it possible to apply `usage` to a `table`?

If not, there should be a spec about what privilege can be applied to each kind of object.

File: docs/doc/60-contributing/03-rfcs/20220809-share.md [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+
+### Create Database, Table FROM Share
+
+After tenants have permission to access the share objects, can create the database from the share.
+
+Syntax:
+
+```sql
+CREATE DATABASE <name> FROM SHARE <provider_tenant>.<share_name>
+```
```
Comment (drmingdrmer) [inline docs/doc/60-contributing/03-rfcs/20220809-share.md:122]: I am not very familiar with the `share`, thus my question is what's the difference between a normal database and one created from a share?

File: docs/doc/60-contributing/03-rfcs/20220809-share.md [hunk 12]
```diff
@@ -0,0 +1,171 @@
+---
+title: Share objects
+description: Share objects between tenants
+---
+
+- Tracking Issue: [Feature: tracing issue of share SQL support · Issue #6973 · datafuselabs/databend](https://github.com/datafuselabs/databend/issues/6973)
```
Comment (Xuanwo) [inline docs/doc/60-contributing/03-rfcs/20220809-share.md:6]:
```suggestion
- Tracking Issue: https://github.com/datafuselabs/databend/issues/6973
```

Use github and docusaurus native link.

File: docs/doc/60-contributing/03-rfcs/20220809-share.md (hunk 9 above)
Comment (lichuang) [reply docs/doc/60-contributing/03-rfcs/20220809-share.md]: yes.

File: docs/doc/60-contributing/03-rfcs/20220809-share.md (hunk 10 above)
Comment (lichuang) [reply docs/doc/60-contributing/03-rfcs/20220809-share.md:203]: has added more comments on privilege.

File: docs/doc/60-contributing/03-rfcs/20220809-share.md (hunk 11 above)
Comment (lichuang) [reply docs/doc/60-contributing/03-rfcs/20220809-share.md:175]: users have no permission to modify data from a database created from a share.

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
It would be better to introduce the share with consumers and producers model.
Like the chapter of `DDL for shares` is producers operation, and `CREATE DATABASE <name> FROM SHARE <provider_tenant>.<share_name>` is comsumer operation.
For a consumer, we should make some questions to be clear:
1. what are the privileges(permission) for the shared database shared by the producer?
2. what will happen to the producer when the consumer `drop the <shared-database>`?
3. what will happen to the consumer if the producer drops the shared database?

**Review by flaneur2020 — APPROVED** [review]
👍👍👍👍

**Comment by Xuanwo** [comment]
> It would be better to introduce the share with consumers and producers model.

Created an issue https://github.com/datafuselabs/databend/issues/7094 to track this.

**Review by drmingdrmer — APPROVED** [review]

**Pull request merged**